src/rocker/nvidia_extension.py

```diff
@@ -138,7 +138,6 @@ def register_arguments(parser, defaults={}):
             choices=['auto', 'runtime', 'gpus'],
             nargs='?',
             const='auto',
-            default=defaults.get(Nvidia.get_name(), 'auto'),
```

Comment:
I think that we need to have a None as the default. Removing this capability prevents the use of the `defaults = {}` map. We basically need a weird state where there's effectively a 4th state which is None. Then if you set --nvidia you have the ternary selection you've defined. The way we've done this in the past is to use a second argument and I think that might make sense to follow that pattern here too. The option `--nvidia` enables the plugin. And use `--nvidia-runtime=[auto, runtime, gpus]` and defaults to `auto`. It requires you to pass both argument to override the runtime, but very few people will want to do that so this will be simpler on average, but enable the power users like yourself.

Reply:
> I think that we need to have a None as the default. 

Before I start could you tell me if the 2nd fix meets package pattern? I'm just testing `test_nvidia.NvidiaTest.test_nvidia_extension_basic` and all kwargs combinations and it's working with default argument now. Otherwise, we will do as you explained.